exercises/isbn-verifier/canonical-data.json

```diff
@@ -0,0 +1,88 @@
+{
+    "exercise": "isbn-verifier",
+    "version": "1.0.0",
+    "comments": [
+        "An expected value of true indicates a valid ISBN-10, ",
+        "whereas false means the ISBN-10 is invalid."
+    ],
+    "cases": [
+        {
+            "description": "valid isbn number",
+            "property": "isbn",
+            "input": "3-598-21508-8",
+            "expected": true
+        },
+        {
+            "description": "invalid isbn check digit",
+            "property": "isbn",
+            "input": "3-598-21508-9",
+            "expected": false
+        },
+        {
+            "description": "valid isbn number with a check digit of 10",
+            "property": "isbn",
+            "input": "3-598-21507-X",
+            "expected": true
+        },
+        {
+            "description": "check digit is a character other than X",
+            "property": "isbn",
+            "input": "3-598-21507-A",
+            "expected": false
+        },
+        {
+            "description": "invalid character in isbn",
+            "property": "isbn",
+            "input": "3-598-2K507-0",
+            "expected": false
+        },
+        {
+            "description": "X is only valid as a check digit",
+            "property": "isbn",
+            "input": "3-598-2X507-0",
+            "expected": false
+        },
+        {
+            "description": "valid isbn without separating dashes",
+            "property": "isbn",
+            "input": "3598215088",
+            "expected": true
+        },
+        {
+            "description": "isbn without separating dashes and X as check digit",
+            "property": "isbn",
+            "input": "359821507X",
+            "expected": true
+        },
+        {
+            "description": "isbn without check digit and dashes",
+            "property": "isbn",
+            "input": "359821507",
+            "expected": false
+        },
+        {
+            "description": "too long isbn and no dashes",
+            "property": "isbn",
+            "input": "3598215078X",
+            "expected": false
+        },
+        {
+            "description": "isbn without check digit",
+            "property": "isbn",
+            "input": "3-598-21507",
+            "expected": false
+        },
+        {
+            "description": "too long isbn",
+            "property": "isbn",
+            "input": "3-598-21507-XA",
```

Comment:
An implementation that would potentially attempt to checksum ISBNs that are too long would still reject this input because it would see that the "A" is invalid.

**If** the intention of this case is to ensure that an implementation does in fact reject ISBNs that are too long, at the very least a valid character should be used instead of A.

If one wanted to be thorough, one might have to include three different cases here, respectively checking for implementations that try to checksum:

* all 11 resulting digits (with a coefficient of 11 on the leftmost one, naturally)
* the 10 rightmost digits
* the 10 leftmost digits

by providing a too-long ISBN that has a valid check digit under each respective scheme.